### PR TITLE
[REVIEW] fix segfault when edgevals not provided

### PR DESCRIPTION
**Pull request by Iroy30**
Fixes issue #164

### CODE REVIEW

File: cpp/src/nvgraph_gdf.cu [hunk 1]
```diff
@@ -278,15 +279,45 @@ gdf_error gdf_balancedCutClustering_nvgraph(gdf_graph* gdf_G,
...
 	// Ensure that the input graph has values
 	GDF_TRY(gdf_add_adj_list(gdf_G));
-	GDF_REQUIRE(gdf_G->adjList->edge_data != nullptr, GDF_INVALID_API_CALL);
+	//GDF_REQUIRE(gdf_G->adjList->edge_data != nullptr, GDF_INVALID_API_CALL);
 
 	// Initialize Nvgraph and wrap the graph
 	nvgraphHandle_t nvg_handle = nullptr;
 	nvgraphGraphDescr_t nvgraph_G = nullptr;
+        cudaDataType_t settype;
+        Vector<double> d_val;
```
Comment (afender) [inline cpp/src/nvgraph_gdf.cu]: Why do we need double instead of float?

File: python/cugraph/spectral_clustering/test_balanced_cut.py [hunk 2]
```diff
@@ -53,6 +53,25 @@ def random_call(G, partitions):
 @pytest.mark.parametrize('graph_file', DATASETS)
 @pytest.mark.parametrize('partitions', PARTITIONS)
 def test_modularity_clustering(graph_file, partitions):
+    # Read in the graph and get a cugraph object
+    M = read_mtx_file(graph_file).tocsr()
+    row_offsets = cudf.Series(M.indptr)
+    col_indices = cudf.Series(M.indices)
+    G = cugraph.Graph()
+    G.add_adj_list(row_offsets, col_indices)
```
Comment (afender) [inline python/cugraph/spectral_clustering/test_balanced_cut.py]: Can we also test on edge list?

File: cpp/src/nvgraph_gdf.cu (hunk 1 above)
Comment (afender) [reply cpp/src/nvgraph_gdf.cu]: Synced up with Ishika onsite, f64 is more stable. Keeping 64 bit by default.
